### PR TITLE
Match SEPA accounts IBANs to selected participating countries #241

### LINKED ISSUE #241: Reset withdrawal from fields after withdrawal
In the funds/withdrawal screen the input fields (amount, addresses) are not reset after withdrawal.
Same for the trade process screen.

### PR DESCRIPTION
**Pull request by alyokaz**
Auto fills bank county code based on a valid IBAN code and validates the IBAN on the selected bank country code.

Fixes #241

Auto-fills the bank country code based on the IBAN entered, conversely makes the selected bank country code part of the validation of the IBAN.

Wasn't too sure on how this should be tested (i.e. testpad), so thought I would leave to see if this is considered the right approach first.

### CODE REVIEW

**Comment by pazza83** [comment]
Hi @AllyKaz and @ripcurlx 

Does this address https://github.com/bisq-network/growth/issues/241 and https://github.com/bisq-network/bisq/issues/5384 ?

**Comment by ripcurlx** [comment]
@AllyKaz Could you please sign your commits. Thanks!

File: desktop/src/main/java/bisq/desktop/components/paymentmethods/SepaForm.java [hunk 1]
```diff
@@ -96,6 +97,8 @@ public void addFormForAddAccount() {
             updateFromInputs();
         });
 
+
+
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/components/paymentmethods/SepaForm.java:101]: Could you please remove this unnecessary blank lines.

File: desktop/src/main/java/bisq/desktop/components/paymentmethods/SepaForm.java [hunk 2]
```diff
@@ -104,6 +107,7 @@ public void addFormForAddAccount() {
             updateFromInputs();
 
         });
+
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/components/paymentmethods/SepaForm.java:110]: Could you please remove this unnecessary blank lines.

File: desktop/src/main/java/bisq/desktop/components/paymentmethods/SepaForm.java [hunk 3]
```diff
@@ -112,6 +116,8 @@ public void addFormForAddAccount() {
 
         });
 
+
+
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/components/paymentmethods/SepaForm.java:120]: Could you please remove this unnecessary blank lines.

File: desktop/src/main/java/bisq/desktop/components/paymentmethods/SepaForm.java [hunk 4]
```diff
@@ -128,6 +134,31 @@ public void addFormForAddAccount() {
...
+                if (!ibanCountryCode.equals(comboBoxCountryCode)) {
+                    InputValidator.ValidationResult result = new InputValidator
+                            .ValidationResult(false, "Country code must match country of bank");
+                    ibanInputTextField.validationResultProperty().setValue(result);
+                }
+            }
+        });
+
+
+
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/components/paymentmethods/SepaForm.java:161]: Could you please remove this unnecessary blank lines.

**Review by ripcurlx — CHANGES_REQUESTED** [review]
NACK - Thanks for picking up this issue! Please see my comments regarding empty lines.

**Comment by cd2357** [comment]
@AllyKaz any updates? Looks like a promising PR.

**Comment by pazza83** [comment]
Please can this be kept open as it is yet to be implemented.

**Comment by ghost** [comment]
@ripcurlx This one could be closed now

**Pull request closed without merging**